libevm/VMOpt.cpp

```diff
@@ -49,20 +49,22 @@ void VM::initMetrics()
 	done = true;
 }
 
-//
-// init interpreter on entry
-//
-
+/// Init interpreter on entry.
 void VM::initEntry()
 {
-	m_bounce = &VM::interpretCases;	
+	m_bounce = &VM::interpretCases;
+
+	// Copy and extend code by 32 bytes to handle virtual push data.
```

Comment:
Comment updated.

reserve() allocates memory, but vector's size is 0, `=` copies one vector data to the second one, resize() fills the additional bytes with zero. We can't do much to make it look better and being optimal.

I also added a VM tests for PUSH32 without anything after it. We don't need 33 bytes because the "virtual STOP" is handled by `VM::fetchInstruction()` in https://github.com/ethereum/cpp-ethereum/blob/develop/libevm/VM.cpp#L110.